internal/pkg/chunker/stream/stream.go

```diff
@@ -64,29 +67,53 @@ func (c *Stream) Read(ctx context.Context) <-chan []byte {
 		// nolint: errcheck
 		defer c.source.Close()
 
+		var (
+			err     error
+			watcher *fsnotify.Watcher
+		)
+
+		if watcher, err = fsnotify.NewWatcher(); err != nil {
+			fmt.Printf("failed to create watcher: %v\n", err)
+			return
+		}
+		if err = watcher.Add(c.source); err != nil {
+			fmt.Printf("watch error: %v\n", err)
+			return
+		}
+
+		// nolint: errcheck
+		defer watcher.Close()
+
 		buf := make([]byte, c.options.Size)
 		for {
 			select {
 			case <-ctx.Done():
 				return
-			default:
-				n, err := c.source.Read(buf)
-				if err != nil {
-					if err != io.EOF {
-						fmt.Printf("read error: %s\n", err.Error())
-						break
+			case e := <-watcher.Events:
```

Comment:
This isn't gonna work. The file won't be read initially until a write. Needs work.